### PR TITLE
chore: update crypto-js to @4 in aws-amplify@3

### PR DESCRIPTION
**Pull request by iartemiev**
#### Description of changes

* Upgrading CryptoJS to 4
* Adding workaround implementation of `WordArray.random` for RN since it doesn't support latest CryptoJS

#### Issue #, if available

https://github.com/aws-amplify/amplify-js/issues/8256

#### Description of how you validated changes
* Built the lib code, deployed to Verdaccio
* Installed from Verdaccio into RN sample apps that use Auth and DataStore
* Validated that expected functionality is intact in those
* Installed into web sample apps for the same categories and validated the same ^
* Ran e2e tests in CCI. Link to successful run: https://app.circleci.com/pipelines/github/aws-amplify/amplify-js/7878/workflows/114cab0a-2236-4a34-aef7-42355eb9cb86

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by sammartinez — APPROVED** [review]
LGTM 🌮

**Review by sammartinez — APPROVED** [review]
LGTM 🌮

File: packages/datastore/src/util.ts [hunk 1]
```diff
@@ -19,6 +19,7 @@ import {
 	SortPredicatesGroup,
 	SortDirection,
 } from './types';
+import { WordArray } from 'amazon-cognito-identity-js';
```
Comment (manueliglesias) [inline packages/datastore/src/util.ts:22]: should this appear now as a dependency in the package.json? (managed by lerna)

File: packages/auth/package.json [hunk 2]
```diff
@@ -44,7 +44,7 @@
     "@aws-amplify/cache": "3.1.56",
     "@aws-amplify/core": "3.8.23",
     "amazon-cognito-identity-js": "4.6.1",
-    "crypto-js": "^3.3.0"
+    "crypto-js": "4.0.0"
```
Comment (manueliglesias) [inline packages/auth/package.json:47]: Here it is defined as an exact version, but on  packages/amazon-cognito-identity-js/package.json is with a caret, let's be consistent

**Review by manueliglesias — APPROVED** [review]
LGTM, two minor comments

File: packages/auth/package.json (hunk 2 above)
Comment (iartemiev) [reply packages/auth/package.json:47]: Good catch. That was a fat finger, I'm about to correct it

File: packages/datastore/src/util.ts (hunk 1 above)
Comment (iartemiev) [reply packages/datastore/src/util.ts:21]: Updated

**Review by manueliglesias — APPROVED** [review]
LGTM 👍

**Review by amhinson — APPROVED** [review]
LGTM 👍

**Review by sammartinez — APPROVED** [review]
LGTM 🌮

**Pull request merged**